### PR TITLE
remove system database reconciliation on external databases mode

### PR DESCRIPTION
**Pull request by eguzki**

### CODE REVIEW

**Comment by miguelsorianod** [comment]
If we want to remove reconciliation of those Secrets the change looks good :+1: . Why we don't want to allow reconciliation of those fields?

File: pkg/3scale/amp/operator/system_reconciler.go [hunk 1]
```diff
@@ -232,31 +232,9 @@ func (r *SystemReconciler) Reconcile() (reconcile.Result, error) {
...
 	}
 
-	// TODO rethink where to create the system-database secret
-	if r.apiManager.Spec.HighAvailability != nil && r.apiManager.Spec.HighAvailability.Enabled {
-		ha, err := r.highAvailability()
-		if err != nil {
-			return reconcile.Result{}, err
-		}
-
-		err = r.reconcileDatabaseHASecret(ha.SystemDatabaseSecret())
```
Comment (miguelsorianod) [inline pkg/3scale/amp/operator/system_reconciler.go]: We could remove the reconcileDatabaseHASecret method itself then too

Comment (eguzki) [reply pkg/3scale/amp/operator/system_reconciler.go]: Templates need to create the secret with parameters passed to the template. Method cannot be removed.

**Comment by eguzki** [comment]
>  Why we don't want to allow reconciliation of those fields?

* reconciliation of database url is not supported
* There are no optional attributes, so reconciliation is not needed to set defaults.

File: pkg/3scale/amp/operator/system_reconciler.go (hunk 1 above)
Comment (miguelsorianod) [reply pkg/3scale/amp/operator/system_reconciler.go]: But the reconcileDatabaseHASecret is an unexported method of the "operator" package (pkg/3scale/amp/operator), it is not used by the templates.

Comment (eguzki) [reply pkg/3scale/amp/operator/system_reconciler.go]: sure, I was referring to `ha.SystemDatabaseSecret()`

**Comment by eguzki** [comment]
ready to review @miguelsorianod

**Pull request merged**